agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/statsbeat/AzureMetadataService.java

```diff
@@ -68,7 +68,7 @@ void scheduleWithFixedDelay(long interval) {
     // Querying Azure Metadata Service is required for every 15 mins since VM id will get updated
     // frequently.
     // Starting and restarting a VM will generate a new VM id each time.
-    scheduledExecutor.scheduleWithFixedDelay(this, interval, interval, TimeUnit.SECONDS);
+    scheduledExecutor.scheduleWithFixedDelay(this, 0, interval, TimeUnit.SECONDS);
```

Comment:
is it important for this to happen right away? would be better to delay it a little bit so it doesn't potentially impact startup time

Reply:
5 sec?